src/bls/bls_aggregator.cpp

```diff
@@ -162,98 +164,104 @@ namespace {
-
-            // NOTE: Check if key is in smart contract but not in Oxen
-            for (size_t index = 0; index < contract_ids.ids.size(); index++) {
-                uint64_t contract_id = contract_ids.ids[index];
-                const bls_public_key& contract_bls_pkey = contract_ids.bls_pubkeys[index];
-                bool found = false;
-                for (const auto& sn_info : sn_list_info) {
-                    if (sn_info.info->bls_public_key == contract_bls_pkey) {
-                        found = true;
-                        break;
-                    }
-                }
-
-                if (!found) {
-                    missing_count++;
-                    oxen::log::warning(
-                            logcat,
-                            "Session node {} {} exists in contract but not in oxen",
-                            contract_id,
-                            contract_bls_pkey);
-                }
-            }
-
-            if (missing_count == 0)
-                oxen::log::debug(logcat, "No missing nodes detected!");
-        }
-
-        // NOTE: Re-derive the BLS aggregate key and the key subtraction step
-        {
-            // NOTE: Aggregate all keys
-            eth::pubkey_aggregator cpp_agg_pubkey;
-            for (size_t index = 0; index < contract_ids.ids.size(); index++)
-                cpp_agg_pubkey.add(contract_ids.bls_pubkeys[index]);
-
-            oxen::log::debug(logcat, "Full BLS aggregate public key {}", cpp_agg_pubkey.get());
-
-            // NOTE: Subtract non-signers
-            for (size_t index = 0; index < contract_ids.ids.size(); index++) {
-                bool found = false;
-                const bls_public_key& contract_bls_pkey = contract_ids.bls_pubkeys[index];
-                for (const bls_public_key& pkey : signature_signers) {
-                    if (pkey == contract_bls_pkey) {
-                        found = true;
-                        break;
-                    }
-                }
-
-                if (!found) {
-                    oxen::log::debug(
-                            logcat,
-                            "  Subtracting BLS key from Session Node {} {}",
-                            contract_ids.ids[index],
-                            contract_bls_pkey);
-                    cpp_agg_pubkey.subtract(contract_bls_pkey);
-                }
-            }
-
-            // NOTE: Dump the key we re-derived (e.g. includes the non-signers)
-            oxen::log::debug(
-                    logcat,
-                    "Re-derived (via subtraction) BLS aggregate public key {}",
-                    cpp_agg_pubkey.get());
-        }
+            cryptonote::core& core, std::unordered_map<bls_public_key, bls_signature> signatures) {
+
+        if (log::get_level(logcat) <= log::Level::debug)
+            core.l2_tracker().get_all_service_node_ids(
+                    std::nullopt,
+                    [&core, signatures = std::move(signatures)](
```

Comment:
```suggestion
            cryptonote::core& core, const std::unordered_map<bls_public_key, bls_signature>& signatures) {

        if (log::get_level(logcat) <= log::Level::debug)
            core.l2_tracker().get_all_service_node_ids(
                    std::nullopt,
                    [&core, signatures](
```

Take by ref and only produce a copy if in the lambda, e.g. in debug logging mode only.